cogs/birthday.py

```diff
@@ -0,0 +1,126 @@
+import discord
+from discord import app_commands
+from discord.ext import tasks, commands
+import config
+import asyncio
+import datetime
+from datetime import date
+import os
+import pytz
+import requests
+
+AIRTABLE_API_KEY = os.environ['UPE_AIRTABLE_API_KEY']
+AIRTABLE_BASE_ID = os.environ['UPE_AIRTABLE_BASE_KEY']        
+AIRTABLE_TABLE_ID = os.environ['UPE_AIRTABLE_TABLE_ID']
+BIRTHDAY_FIELD_ID = 'fldoUSAvFtdsI51FP'
+DISCORD_NAME_FIELD_ID = 'fldoUSAvFtdsI51FP'
+NAME_FIELD_ID = 'fld3mtI8hayU9UPHD'
+YELLOW_COLOR = 0xFFBF00  
+RED_COLOR = 0xD2222D
+
+BOT_KEY = os.environ['BOT_KEY']
+
+class Birthday(commands.Cog):
+    '''
+    Checks 8am every day for birthdays
+    If there are birthdays @marketing
+    '''
+
+    def __init__(self, bot):
+        self.bot = bot
+        self.MEMBERSHIP_CHANNEL_ID = 1061194622472298559 if config.isProd else 1065042160426176632 
+        self.BOT_LOG_CHANNEL_ID = 626541886533795850 if config.isProd else 1065042159679578154
+        self.SPOTLIGHT_CHANNEL_ID = 1019725350701379685
+        self.GUILD = bot.get_guild(245393533391863808)
+        self.check_birthday.start()
+    
+    def unload(self):
+        self.check_birthday.cancel()
+
+    def get_airtable_data(self):
+        '''
+        Returns full response json, including pages
+        '''
+
+        all_records = []
+        endpoint = f'https://api.airtable.com/v0/{AIRTABLE_BASE_ID}/{AIRTABLE_TABLE_ID}'
+        headers = {
+            'Authorization': f'Bearer {AIRTABLE_API_KEY}', 
+            'Content-Type': 'application/json'
+        }
+        url = endpoint
+        while True:
+            try:
+                response = requests.get(url, headers=headers)
+                response.raise_for_status()
+                data = response.json()
+                records = data['records']
+                all_records.extend(records)
+                if 'offset' in data:
+                    url = endpoint + '?offset=' + data['offset']
+                else:
+                    break
+            except Exception as e:
+                print(f'Something went wrong with the airtable query: {e}')
+        
+        return all_records
+
+    @tasks.loop(minutes=59)
+    async def check_birthday(self):
+        '''
+        Checks json for birthday, if match, send message with discord names
+        '''
+        now = datetime.datetime.now()
+        current_date = now.strftime("%m-%d")
+        date_formatted = now.strftime('%B %d') + ('st' if now.day in [1, 21, 31] else 'nd' if now.day in [2, 22] else 'rd' if now.day in [3, 23] else 'th')
+        
+        if not now.hour == 8:
+            return
```

Comment:
This isn't necessary if you supply a datetime object to the loop decorator above. That way, it'll always run at the same time every day instead of every X interval after startup

Refer to the `time` parameter under `@discord.ext.tasks.loop` https://discordpy.readthedocs.io/en/v2.1.0/ext/tasks/index.html#discord.ext.tasks.loop

Reply:
Changed this and added instead the time parameter!

`TIME_TO_RUN = time(hour=8, tzinfo=pytz.timezone('US/Eastern'))`
`@tasks.loop(time=TIME_TO_RUN)`